Automattic/wp-calypso
# Enable SSR for OAuth2 native login

Fixes https://github.com/Automattic/wp-calypso/issues/14159 (Previous attempt https://github.com/Automattic/wp-calypso/pull/16133) and enables SSR for OAuth login flows in calypso.

Note: Printing the data of the oauth2 client is not done yet as the fetch query cannot be done on the server. I prepared [another PR](https://github.com/Automattic/wp-calypso/pull/17166) for this which requires to move the client data to the redux state as initial state. I did not add it here as this PR is already big enough to review.

### Testing Instructions
- Boot this branch locally `git checkout add/oauth-login-ssr && npm start`
- Run the following command in your terminal and check that there is a matching for it:
```
curl 'http://calypso.localhost:3000/log-in?client_id=930' | grep "vaultpress"
```
- Load this url in your browser while logged out and check that the layout (masterbar) does not change while the page is loading anymore

### Reviews
